javascript/sources/entry-common.js

```diff
@@ -129,6 +83,77 @@ module.exports = function (bind, lib) {
 
     }
 
+    class Value {
+
+        constructor(unit, value) {
+
+            this.unit = unit;
+            this.value = value;
+
+        }
+
+        fromJS(expose) {
+
+            expose(this.unit, this.value);
+
+        }
+
+        toString() {
+
+            switch (this.unit) {
+
+                case constants.UNIT_PIXEL:
+                    return `${this.value}`;
+
+                case constants.UNIT_PERCENT:
+                    return `${this.value}%`;
+
+                default: {
+                    return `${this.value}?`;
+                }
+
+            }
+
+        }
+
+        valueOf() {
+
+            return this.value;
+
+        }
+
+    }
+
+    for (let fnName of [ `setPosition`, `setMargin`, `setFlexBasis`, `setWidth`, `setHeight`, `setMinWidth`, `setMinHeight`, `setMaxWidth`, `setMaxHeight`, `setPadding` ]) {
+
+        let methods = { [constants.UNIT_PIXEL]: lib.Node.prototype[fnName], [constants.UNIT_PERCENT]: lib.Node.prototype[`${fnName}Percent`] };
```

Comment:
nit: could we throw an exception here if either are undefined. It would mean that you would know early on if you'd not fully implements support for Percent on a new function and all tests would fail